Userland/Utilities/js.cpp

```diff
@@ -1653,7 +1654,16 @@ ErrorOr<int> serenity_main(Main::Arguments arguments)
                 auto file = TRY(Core::File::open(path, Core::OpenMode::ReadOnly));
                 auto file_contents = file->read_all();
                 auto source = StringView { file_contents };
-                builder.append(source);
+
+                if (Utf8View { file_contents }.validate()) {
+                    builder.append(source);
+                } else {
+                    auto* decoder = TextCodec::decoder_for("windows-1252");
```

Comment:
Wait, who says the input is CP-1252 encoded? What about anything other than UTF-8 and that?

Reply:
There isn't really a way to determine that. If there's a BOM, of course we can detect that and decode it properly. Otherwise, the editors / REPLs I checked seemed to just use a Latin-1 codec as a default.

Alternatively, we could just bail here if there isn't a BOM

Reply:
Alright :)